### PR TITLE
Improve histogram and barplot viz controls

### PR DESCRIPTION
**Pull request by bobular**
Depends on https://github.com/VEuPathDB/web-components/pull/169 

So far I have done

- fixed the binWidth control to handle date units
- added absolute/relative control to histogram
- added absolute/relative and logscale controls to barplot

I've chosen not to add zooming controls to histogram x-axis at this time.
Maybe wait until after #98 to add this, if still wanted.

### CODE REVIEW

**Comment by bobular** [comment]
I've added barplot controls (log and absolute/relative) too.

The issue below (>100% proportion) will be solved when we roll out the new barmode options in #246 
image

**Comment by bobular** [comment]
All good now with the latest back end. Ready for review please.

image

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx [hunk 1]
```diff
@@ -301,47 +337,60 @@ function BarplotViz(props: Props) {
...
+            state={dependentAxisLogScale}
+            onStateChange={onDependentAxisLogScaleChange}
+          />
+          <RadioButtonGroup
+            selectedOption={
+              valueSpec === 'proportion' ? 'proportional' : 'count'
+            }
+            options={['count', 'proportional']}
+            onOptionSelected={(newOption) => {
+              if (newOption === 'proportional') {
```
Comment (bobular) [inline src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:385]: Some of this EDA code might be simplified if the RadioButtonGroup component took two props: `displayLabels` and `values`.

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx [hunk 2]
```diff
@@ -48,16 +50,28 @@ function FullscreenComponent(props: VisualizationProps) {
...
-  overlayVariable: VariableDescriptor,
-  facetVariable: VariableDescriptor,
-});
+const BarplotConfig = t.intersection([
+  t.type({
+    dependentAxisLogScale: t.boolean,
+    valueSpec: ValueSpec,
+  }),
+  t.partial({
+    xAxisVariable: VariableDescriptor,
```
Comment (dmfalke) [inline src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:70]: Is `xAxisVariable` really optional?

**Review by dmfalke — COMMENTED** [review]
I've looked at the code, and all looks good. I left an inline comment about one questionable thing.

**Comment by dmfalke** [comment]
@bobular can you fix the compile issues and merge the latest from `main`? Thanks!

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx (hunk 2 above)
Comment (bobular) [reply src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:70]: Yes. When you first open up a new Barplot - there aren't any variables selected by the user.

**Comment by bobular** [comment]
> 
> 
> @bobular can you fix the compile issues and merge the latest from `main`? Thanks!

Hi @dmfalke - I did this earlier today but forgot to push.
It will definitely compile alongside VEuPathDB/web-components#169

**Review by dmfalke — APPROVED** [review]

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx (hunk 2 above)
Comment (dmfalke) [reply src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:70]: Oh yeah... :upside_down_face:

**Pull request merged**